rust/xaynet-analytics/src/database/isar.rs

```diff
@@ -90,32 +81,22 @@ impl IsarDb {
         Ok(self.get_collection(collection_name)?.new_string_oid(oid))
     }
 
-    // TODO: move this logic to the Repo within the AnalyticsController ticket: https://xainag.atlassian.net/browse/XN-1559
-    fn get_schema() -> Result<Schema, Error> {
-        let mut schema = Schema::new();
-        schema
-            .add_collection(get_collection_schema(
-                Self::ANALYTICS_EVENT_COLLECTION_NAME,
-                &mut AnalyticsEvent::into_field_properties(),
-            )?)
-            .map_err(|_| {
-                anyhow!(
-                    "failed to add collection {} to schema",
-                    Self::ANALYTICS_EVENT_COLLECTION_NAME
-                )
-            })?;
-        schema
-            .add_collection(get_collection_schema(
-                Self::SCREEN_ROUTE_COLLECTION_NAME,
-                &mut ScreenRoute::into_field_properties(),
-            )?)
-            .map_err(|_| {
-                anyhow!(
-                    "failed to add collection {} to schema",
-                    Self::SCREEN_ROUTE_COLLECTION_NAME
-                )
-            })?;
-        Ok(schema)
+    pub fn dispose(self) -> Result<(), Error> {
+        match self.instance.close() {
+            Some(_) => Err(anyhow!("err")),
```

Comment:
the `instance.close()` returns the instance if it couldn't be closed. is it ok to discard the instance then with an error?

Reply:
I have no idea what the Isar maintainer has in mind, and there's no documentation. From our point of view, if we want to close it and Isar doesn't / cannot, we should return an error..? (i'll write a proper message as Kwok already pointed out). If it does close it, then we return Ok(). I'm used to the Hive implementation in Dart, which returns `void` in case of success. Do you have other ideas?

Side note: I still have to figure out how to handle errors propagated by the controller. Haven't got that far yet.

Reply:
if i get the `isar` code right, this error case usually shouldn't happen to us, so an error message should be fine for now.

returning `Result<(), ErrorType>` is basically the rust equivalent on returning void on success.

it seems that `isar` internally manages a global dict with all databases. when we create a new instance, we get a handle to the instance and a copy of the handle is stored in that global dict. as long as we don't copy our handle, but only use references to it, things should shut down nicely.